### PR TITLE
magicApe

### PR DESCRIPTION
**Pull request by 0xCalibur**

### CODE REVIEW

File: src/swappers/MagicApeLevSwapper.sol [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import "BoringSolidity/interfaces/IERC20.sol";
+import "BoringSolidity/libraries/BoringERC20.sol";
+import "libraries/SafeApprove.sol";
+import "interfaces/IBentoBoxV1.sol";
+import "interfaces/ILevSwapperV2.sol";
+import "interfaces/IERC4626.sol";
+import "interfaces/IGmxGlpRewardRouter.sol";
+import "interfaces/IGmxVault.sol";
+
+contract MagicApeLevSwapper is ILevSwapperV2 {
```
Comment (0xm3rlin) [inline src/swappers/MagicApeLevSwapper.sol]: One nit pick here, why don't we make this an arbitrary ERC4626 swapper

File: src/swappers/MagicApeLevSwapper.sol [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    using BoringERC20 for IERC20;
+    using SafeApprove for IERC20;
+
+    error ErrSwapFailed();
+
+    IBentoBoxV1 public immutable bentoBox;
+    IERC20 public immutable mim;
+    IERC20 public immutable ape;
```
Comment (0xm3rlin) [inline src/swappers/MagicApeLevSwapper.sol]:
```suggestion
    IERC20 public immutable asset;
```

File: src/swappers/MagicApeLevSwapper.sol [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    IERC4626 public immutable magicApe;
```
Comment (0xm3rlin) [inline src/swappers/MagicApeLevSwapper.sol:23]:
```suggestion
    IERC4626 public immutable vault;
```

**Review by 0xm3rlin — CHANGES_REQUESTED** [review]
Looks good

File: src/swappers/MagicApeLevSwapper.sol (hunk 1 above)
Comment (0xCalibur) [reply src/swappers/MagicApeLevSwapper.sol]: you're right, I'll adapt it so it's generic

**Review by 0xm3rlin — APPROVED** [review]

**Pull request merged**